### PR TITLE
Document how to monitor Argo CD custom resource workloads

### PR DESCRIPTION
**Pull request by Srivaralakshmi**
[RHDEVDOCS-4959](https://issues.redhat.com/browse/RHDEVDOCS-4959): Document how to configure/enable alerts for each workload and how users can expect to receive these alerts

- **Aligned team**: Dev Tools
- **OCP version for cherry-picking**: enterprise-`4.10` and later
- **JIRA issues**: [RHDEVDOCS-4959](https://issues.redhat.com/browse/RHDEVDOCS-4959)
- **Preview pages**: [Monitoring Argo CD custom resource workloads](https://56235--docspreview.netlify.app/openshift-enterprise/latest/cicd/gitops/monitoring-argo-cd-custom-resource-workloads.html)
- **SME Review**: Completed by @jaideepr97
- **QE review**: Completed by @varshab1210
- **Peer-review**:  Completed by @gabriel-rh

### CODE REVIEW

**Comment by Srivaralakshmi** [comment]
@jaideepr97 and @varshab1210: PTAL at the PR and provide feedback. Thanks!

File: cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+With {gitops-title}, you can monitor the availability of Argo CD custom resource workloads for specific Argo CD instances. By monitoring Argo CD custom resource workloads, you have the latest information about the state of your Argo CD instances by enabling alerts for them. The Operator then triggers the alerts when the component workload pods such as application-controller, repo-server, or server of the correpsonding Argo CD instance are unable to come up for certain reasons and there is a drift between the number of ready replicas and the number of desired replicas for a certain period of time.
+
+You can enable and disable the setting for monitoring Argo CD custom resource workloads. 
+
+.Prerequisites
+
+* {gitops-title} is installed in your cluster.
+* Logged into Argo CD instance.
+* Prometheus is configured in your cluster and the Argo CD instance is in a namespace that you can monitor through Prometheus.
+* The `kube-state-metrics` service running in your cluster.
```
Comment (jaideepr97) [inline cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc:19]: @Srivaralakshmi can we add another prerequisite here:

If monitoring is being enabled in an Argo CD instance in a namespace that is _not_ watched by the default OpenShift monitoring stack (i.e any namespace that does not start with "openshift-*") then userworkload monitoring needs to be enabled on the cluster, so that the monitoring stack can pick up the created prometheusRule

**Comment by Srivaralakshmi** [comment]
@jaideepr97 and @varshab1210: Incorporated the suggestions. PTAL at the revised PR and provide feedback. Thanks!

File: cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc (hunk 1 above)
Comment (Srivaralakshmi) [reply cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc]: @jaideepr97  I have addressed your suggestion. PTAL. If all is ok, then pls provide LGTM. Thnx!

**Comment by varshab1210** [comment]
LGTM
Thanks!

File: cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+You can enable and disable the setting for monitoring Argo CD custom resource workloads. 
+
+// Prerequisites for monitoring Argo CD custom resource workloads
+[discrete]
+== Prerequisites
+
+* {gitops-title} is installed in your cluster.
+* Logged into Argo CD instance.
+* Prometheus is configured in your cluster and the Argo CD instance is in a namespace that you can monitor through Prometheus.
```
[the remark this answers is not in the document]
Comment (jaideepr97) [reply cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc:20]: @Srivaralakshmi

**Review by jaideepr97 — CHANGES_REQUESTED** [review]

File: cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (Srivaralakshmi) [reply cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc:20]: @jaideepr97 Addressed. PTAL at the changes.

**Comment by Srivaralakshmi** [comment]
@jaideepr97: Thanks for the feedback! I have incorporated the suggestions. PTAL at the revised PR and provide feedback.

**Comment by jaideepr97** [comment]
/lgtm 
thanks

**Review by jaideepr97 — APPROVED** [review]

File: cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc [hunk 3]
```diff
@@ -0,0 +1,38 @@
+:_content-type: ASSEMBLY
+[id="monitoring-argo-cd-custom-resource-workloads"]
+= Monitoring Argo CD custom resource workloads
+include::_attributes/common-attributes.adoc[]
+:context: monitoring-argo-cd-custom-resource-workloads
+
+toc::[]
+
+[role="_abstract"]
+With {gitops-title}, you can monitor the availability of Argo CD custom resource workloads for specific Argo CD instances. By monitoring Argo CD custom resource workloads, you have the latest information about the state of your Argo CD instances by enabling alerts for them. The Operator then triggers the alerts when the component workload pods such as application-controller, repo-server, or server of the correpsonding Argo CD instance are unable to come up for certain reasons and there is a drift between the number of ready replicas and the number of desired replicas for a certain period of time.
```
Comment (gabriel-rh) [inline cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc:10]: The last sentence here is hard to parse when reading it. Maybe simple re-organizing it may help:

 When the component workload pods such as application-controller, repo-server, or server of the correpsonding Argo CD instance are unable to come up for certain reasons and there is a drift between the number of ready replicas and the number of desired replicas for a certain period of time, the Operator then triggers the alerts.
 
 Or split it up into multiple sentences...

Comment (gabriel-rh) [inline cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc:10]:
```suggestion
With {gitops-title}, you can monitor the availability of Argo CD custom resource workloads for specific Argo CD instances. By monitoring Argo CD custom resource workloads, you have the latest information about the state of your Argo CD instances by enabling alerts for them. The Operator then triggers the alerts when the component workload pods such as application-controller, repo-server, or server of the corresponding Argo CD instance are unable to come up for certain reasons and there is a drift between the number of ready replicas and the number of desired replicas for a certain period of time.
```

File: cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+You can enable and disable the setting for monitoring Argo CD custom resource workloads. 
+
+// Prerequisites for monitoring Argo CD custom resource workloads
+[discrete]
+== Prerequisites
+
+* {gitops-title} is installed in your cluster.
+* Logged into Argo CD instance.
```
Comment (gabriel-rh) [inline cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc:19]: This should probably be a full sentence, to match the others, eg

You are logged into...

File: modules/gitops-disabling-monitoring-for-argo-cd-custom-resource-workloads.adoc [hunk 5]
```diff
@@ -0,0 +1,30 @@
+// Module included in the following assemblies:
+//
+// * gitops-argo-cd-installation.adoc
```
Comment (gabriel-rh) [inline modules/gitops-disabling-monitoring-for-argo-cd-custom-resource-workloads.adoc:3]: needs a path here for this file

File: modules/gitops-enabling-monitoring-for-argo-cd-custom-resource-workloads.adoc [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+[source,yaml]
+----
+apiVersion: argoproj.io/v1alpha1
+kind: ArgoCD
+metadata:
+  name: example-argocd
+  labels:
+    example: repo
+spec:
+   ...
```
Comment (gabriel-rh) [inline modules/gitops-enabling-monitoring-for-argo-cd-custom-resource-workloads.adoc:28]: I would align these dots one position to the left

File: modules/gitops-disabling-monitoring-for-argo-cd-custom-resource-workloads.adoc [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+[source,yaml]
+----
+apiVersion: argoproj.io/v1alpha1
+kind: ArgoCD
+metadata:
+  name: example-argocd
+  labels:
+    example: repo
+spec:
+   ...
```
Comment (gabriel-rh) [inline modules/gitops-disabling-monitoring-for-argo-cd-custom-resource-workloads.adoc:26]: I would align these dots one position to the left

File: modules/gitops-enabling-monitoring-for-argo-cd-custom-resource-workloads.adoc [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+          expr: >-
+            kube_statefulset_status_replicas{statefulset="openshift-gitops-application-controller statefulset",
+            namespace="openshift-gitops"} !=
+            kube_statefulset_status_replicas_ready{statefulset="openshift-gitops-application-controller statefulset",
+            namespace="openshift-gitops"} 
+          for: 1m
+          labels:
+            severity: critical
+----
+<1> Alert rule in the PrometheusRule that checks whether the workloads created by the Argo CD instances are running as expected.
```
Comment (gabriel-rh) [inline modules/gitops-enabling-monitoring-for-argo-cd-custom-resource-workloads.adoc:64]: maybe tick PrometheusRule

File: modules/gitops-enabling-monitoring-for-argo-cd-custom-resource-workloads.adoc [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  labels:
+    example: repo
+spec:
+   ...
+  monitoring:
+    enabled: true
+  ...
+----
+
+. Verify whether an alert rule is included in the PrometheusRule created by the Operator:
```
Comment (gabriel-rh) [inline modules/gitops-enabling-monitoring-for-argo-cd-custom-resource-workloads.adoc:34]: maybe tick PrometheusRule

**Review by gabriel-rh — COMMENTED** [review]
looks good - some minor suggestions

File: cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc (hunk 4 above)
Comment (jaideepr97) [reply cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc:19]: @Srivaralakshmi On second thought, being logged into Argo CD instance is not really required, we can probably get rid of this point 
As long as users are logged into their OpenShift cluster that is enough for this feature

File: modules/gitops-enabling-monitoring-for-argo-cd-custom-resource-workloads.adoc (hunk 8 above)
Comment (Srivaralakshmi) [reply modules/gitops-enabling-monitoring-for-argo-cd-custom-resource-workloads.adoc:64]: @gabriel-rh not required here

File: modules/gitops-enabling-monitoring-for-argo-cd-custom-resource-workloads.adoc (hunk 9 above)
Comment (Srivaralakshmi) [reply modules/gitops-enabling-monitoring-for-argo-cd-custom-resource-workloads.adoc:34]: @gabriel-rh not required here

File: cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc (hunk 3 above)
Comment (Srivaralakshmi) [reply cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc:10]: Addressed it.

Comment (Srivaralakshmi) [reply cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc:10]: Addressed it.

File: cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc (hunk 4 above)
Comment (Srivaralakshmi) [reply cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc:19]: @gabriel-rh @jaideepr97 Removed this point.

**Comment by Srivaralakshmi** [comment]
@gabriel-rh I have incorporated your suggestions. PTAL and provide feedback. Thanks!

File: cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc [hunk 10]
```diff
@@ -7,7 +7,7 @@ include::_attributes/common-attributes.adoc[]
...
+With {gitops-title}, you can monitor the availability of Argo CD custom resource workloads for specific Argo CD instances. By monitoring Argo CD custom resource workloads, you have the latest information about the state of your Argo CD instances by enabling alerts for them. When the component workload pods such as application-controller, repo-server, or server of the correspsonding Argo CD instance are unable to come up for certain reasons and there is a drift between the number of ready replicas and the number of desired replicas for a certain period of time, the Operator then triggers the alerts.
```
Comment (gabriel-rh) [inline cicd/gitops/monitoring-argo-cd-custom-resource-workloads.adoc:10]: correspsonding -> corresponding   (stil an extra `s` in there)

**Review by gabriel-rh — COMMENTED** [review]
looks good to me! apart from the single typo

**Comment by Srivaralakshmi** [comment]
@gabriel-rh PTAL at the revised PR and approve. Thanks!

**Review by gabriel-rh — APPROVED** [review]
/lgtm
